model/messagedeletions.go

```diff
@@ -0,0 +1,31 @@
+package model
+
+// MessageDeletions holds information about deleted messages.
+type MessageDeletions struct {
+	Messages []*Message
+}
+
+// ToExternal converts the event into an external representation.
+func (del MessageDeletions) ToExternal() interface{} {
```

Comment:
I wrote this after writing the text below. 

I think I'm now against supporting different versions of the websocket. I think it makes things much easier. The android app could easily support both version because the message event doesn't get breaking changes after we have a structure with discriminators https://github.com/gotify/server/pull/270/commits/59b3b38675e9b47e79ac091d4b54da628c7a31ea#r380298148

Changes to the api are pretty rare so maybe it is just overkill to support multiple versions, also there probably aren't many clients that use the stream api. 

>  Also, once a "version" is no longer needed (assuming clients will adopt the new version), we can simple remove one file and remove the mapping to it.

https://github.com/gotify/server/pull/270#discussion_r380301203
If the decoders will be deleted the major version of gotify/server has to be increased as it is a breaking change. So why not doing this now :D.

Also the change in the api is not severe, the easy fix for clients is to ignore all events which doesn't have the type message.

-----

Only for clarification. This is related to https://github.com/gotify/server/pull/270#discussion_r380292751

If we'd change the MessageDeletions in v3 to return whole messages. ToExternal isn't that nice anymore because it either has to be changed or another method like ToExternalV3 has to be added. Or the api version could be added as param.

Here some example code:
```go
func (del MessageDeletions) ToExternal(apiVersion uint) (interface{}, error) {
	switch apiVersion {
	case 1:
		return nil, ErrHiddenEvent
	case 2:
		res := &MessageDeletionsExternal{
			make([]uint, len(del.Messages)),
		}
		for key, value := range del.Messages {
			res.IDs[key] = value.ID
		}
		return &EventMessageV2{
			Type: "...",
			Content: res
		}
	case 3:
		res := &MessageDeletionsExternalV3{
			Messages: ConvertMessages(del.Messages),
		}
		return &EventMessageV3{
			Type: "...",
			Content: res
		}
	default:
		return nil, ErrUnsupportedApiVersion
	}
}
```
In the stream writer we can then check if the event != and then just send it without going through the decoder.

Reply:
As for the suggestion in the latter part of you comment I think you're right. I only put the `ToExternal` methods outside of the handler as I didn't see a good reason for moving it just now. You're right that new versions of the handler would have to write their own conversion function, so we'd have to move the ToExternal into the handler (as the handler should take care of distinguishing client versions).

Now, do I get it right that you want me to "remove" the v1 handler, so we only keep the v2 one (using discriminators)? I think this would make things much easier to read, and we get rid of the additional layer of handling client versions.

Reply:
Yes, only keep version 2 and remove the version handling altogether.